Firebase/Messaging/FIRMessagingRemoteNotificationsProxy.m

```diff
@@ -421,6 +425,13 @@ - (void)application:(UIApplication *)application
                           @"application:didFailToRegisterForRemoteNotificationsWithError: %@",
                           error.localizedDescription);
 }
+#endif
+
+#if TARGET_OS_OSX
+- (void)application:(NSApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
```

Comment:
`GULApplication` should be used instead of `NSApplication`. Then `#if TARGET_OS_OSX` should not be required.